File: android/app/src/main/java/com/alamada/childmonitor/models/Child.java
package com.alamada.childmonitor.models;

import com.google.gson.annotations.SerializedName;

public class Child {

    @SerializedName("id")                public int    id;
    @SerializedName("first_name")        public String firstName;
    @SerializedName("last_name")         public String lastName;
    @SerializedName("middle_name")       public String middleName;
    @SerializedName("full_name")         public String fullName;
    @SerializedName("age")               public int    age;
    @SerializedName("date_of_birth")     public String dateOfBirth;
    @SerializedName("gender")            public String gender;
    @SerializedName("address")           public String address;
    @SerializedName("guardian_name")     public String guardianName;
    @SerializedName("guardian_contact")  public String guardianContact;
    @SerializedName("diagnosis")         public String diagnosis;
    @SerializedName("diagnosis_notes")   public String diagnosisNotes;
    @SerializedName("status")            public String status;
    @SerializedName("photo")             public String photo;

    /** Returns display name: fullName if available, else first + last */
    public String getDisplayName() {
        if (fullName != null && !fullName.isEmpty()) return fullName;
        return (firstName != null ? firstName : "") + " " + (lastName != null ? lastName : "");
    }
}

File: backend/app/Models/Child.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Child extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        'first_name',
        'last_name',
        'middle_name',
        'date_of_birth',
        'gender',
        'address',
        'guardian_name',
        'guardian_contact',
        'diagnosis',
        'diagnosis_notes',
        'photo',
        'status',
        'staff_id',
        'parent_id',
    ];

    protected $casts = [
        'date_of_birth' => 'date',
    ];

    // ── Computed ──────────────────────────────────────────────────

    public function getFullNameAttribute(): string
    {
        return trim("{$this->first_name} {$this->middle_name} {$this->last_name}");
    }

    public function getAgeAttribute(): int
    {
        return $this->date_of_birth->age;
    }

    // ── Relationships ─────────────────────────────────────────────

    public function staff()
    {
        return $this->belongsTo(User::class, 'staff_id');
    }

    public function parent()
    {
        return $this->belongsTo(User::class, 'parent_id');
    }

    public function enrollments()
    {
        return $this->hasMany(Enrollment::class);
    }

    public function currentEnrollment()
    {
        return $this->hasOne(Enrollment::class)->latestOfMany();
    }

    public function attendances()
    {
        return $this->hasMany(Attendance::class);
    }

    public function activities()
    {
        return $this->hasMany(Activity::class);
    }
}
